internal/pkg/config/regexp/regexp.go

```diff
@@ -28,7 +28,7 @@ type Regexp struct {
 
 // MatchString delegates matching to the regex package.
 func (mr *Regexp) MatchString(s string) bool {
-	return mr.r == nil || mr.r.MatchString(s)
+	return mr == nil || mr.r == nil || mr.r.MatchString(s)
```

Comment:
Is this just for safety? I don't think we need this.

Reply:
It was a consequence of changing fields from `regexp.Regexp` to `*regexp.Regexp`.  Because nil values can still have interface data, it is possible to call `(*Regexp)(nil).MatchString`, which blew the stack here.

Might not be necessary now that the matchers are polymorphic... will investigate.